daemons/gptp/common/wltimestamper.hpp

```diff
@@ -83,18 +83,8 @@ class WirelessDialog {
 
 };
 
-inline uint64_t _32to64(uint32_t *rollover, uint32_t *prev, uint32_t curr) {
-	uint64_t ret;
-	if (curr < *prev && (*prev - curr) >(UINT32_MAX >> 1)) {
-		*rollover = *rollover + 1;
-	}
-	ret = *rollover;
-	ret <<= 32;
-	ret += curr;
-	*prev = curr;
-	return ret;
-}
-
+#define MAX_NSEC (42949672960)
```

Comment:
ULONG_MAX would be 4294967295 or 0xFFFF_FFFFU, so MAX_NSEC (42949672960, or 0x0000_000A_0000_0000ULL) is 10*(ULONG_MAX + 1).  The parameter being passed to it is being multiplied by 10, so it's possibly a bugfix, but with the weird MAX_NSEC and the multiplication by 10 and the _32to64(uint64_t) function, the section could stand some refactoring or at least explanatory comments.